morpheus/_lib/include/morpheus/messages/control.hpp

```diff
@@ -180,7 +180,7 @@ struct ControlMessageProxy
      * @param value
      */
     static void set_metadata(ControlMessage& self, const std::string& key, pybind11::object& value);
-    static pybind11::object get_metadata(ControlMessage& self, const std::string& key);
+    static pybind11::object get_metadata(ControlMessage& self, std::optional<std::string> const& key);
```

Comment:
This should return a `pybind11::dict`

Reply:
unfortunately this function can return either a `dict` or the value pointed to by the metadata key, so I'm not sure we can change this.